src/managers/GuildMemberRoleManager.js

```diff
@@ -47,6 +47,17 @@ class GuildMemberRoleManager extends DataManager {
     return hoistedRoles.reduce((prev, role) => (!prev || role.comparePositionTo(prev) > 0 ? role : prev));
   }
 
+  /**
+   * The role of the member used to set their role icon
+   * @type {?Role}
+   * @readonly
+   */
+  get icon() {
+    const iconRoles = this.cache.filter(role => role.icon || role.unicodeEmoji);
+    if (!iconRoles.size) return null;
+    return iconRoles.reduce((prev, role) => (!prev || role.comparePositionTo(prev) > 0 ? role : prev));
```

Comment:
This code is duplicated a few times in this file, might be worth making a function for it instead